webui/react/src/components/SplitPane.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import css from './SplitPane.module.scss';
+
+interface Props {
+  children: [React.ReactElement, React.ReactElement];
+  initialWidth?: number;
+  onChange?: (width: number) => void;
+  open?: boolean;
+}
+
+const SplitPane: React.FC<Props> = ({ children, initialWidth, onChange, open = true }: Props) => {
+  const [isDragging, setIsDragging] = useState(false);
+  const [width, setWidth] = useState(initialWidth ?? 400);
+  const container = useRef<HTMLDivElement>(null);
+  const handle = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const c = (e: MouseEvent) => {
+      e.preventDefault();
+      setIsDragging(true);
+    };
+    const handleRef = handle.current;
+    handleRef?.addEventListener('mousedown', c);
+
+    return () => handleRef?.removeEventListener('mousedown', c);
+  }, []);
+
+  useEffect(() => {
+    if (!isDragging) return;
+    const c = (e: MouseEvent) => {
+      if (!container.current) return;
+
+      e.preventDefault();
+
+      // Get offset
+      const containerOffsetLeft = container.current.getBoundingClientRect().left;
```

Comment:
Something to look into, but not needed for merge:
`getBoundingClientRect` forces the browser to paint the element, which is costly. Given this is called during mousemove, we could probably improve the performance somewhat by getting the clientRectangle from `useResize` which only calls it when the element is resized. Alternatively, we could use our own ResizeObserver and get the `clientRect` from there -- I think the reason why `useResize` isn't doing that is because the `clientRect` uses the `content-box` model, not the `border-box` model, but this element has no margin or padding, so it should be alright?